### ISSUE TITLE
Openshift - Containerised model & CEPH RBD

### ISSUE THREAD

**Issue by jameswilkins** [openshift/origin#9904]
Hi,

I'm unable to successfully use CEPH RBD's as PV/PVCS when we're using the containisered model of Openshift.  I'm not sure whether this is a config issue and/or whether its just un-supported.

If I do the RBD map manually from the host OS, this works:

[root@node-10-44-203-75 ~]# rbd map -p cold-storage ceph047
/dev/rbd0
[root@node-10-44-203-75 ~]# rbd unmap /dev/rbd0

If i drop into the origin-node container:

[root@node-10-44-203-75 ~]# docker exec -it origin-node /bin/bash
[root@node-10-44-203-75 origin]# rbd map -p cold-storage ceph047
rbd: add failed: (30) Read-only file system

When attempting to deploy a container using said storage you get the errors shown below in the log section.
##### Version

oc v1.3.0-alpha.2
kubernetes v1.3.0-alpha.1-331-g0522e63
##### Steps To Reproduce

1) Deploy Openshift with containizered=true via Ansible
2) Have CEPH Cluster ready, and create necessary PV/PVCS
3) Attempt to use PVC as persistent storage
##### Current Result

You get the following in the log files of the node:

Jul 18 16:29:22 node-10-44-XXX-YYY origin-node: I0718 16:29:22.003020   24101 rbd.go:89] ceph secret info: key/<redacted>
Jul 18 16:29:23 node-10-44-XXX-YYY origin-node: I0718 16:29:23.008416   24101 rbd_util.go:229] rbd: map mon 10.44.XXX.YYY:6789
Jul 18 16:29:23 node-10-44-XXX-YYY origin-node: I0718 16:29:23.054314   24101 rbd_util.go:240] rbd: map error exit status 30 rbd: add failed: (30) Read-only file system
Jul 18 16:29:23 node-10-44-XXX-YYY origin-node: I0718 16:29:23.054353   24101 rbd_util.go:229] rbd: map mon 10.44.XXX.YYY:6789
Jul 18 16:29:23 node-10-44-XXX-YYY origin-node: I0718 16:29:23.099929   24101 rbd_util.go:240] rbd: map error exit status 30 rbd: add failed: (30) Read-only file system
Jul 18 16:29:23 node-10-44-XXX-YYY origin-node: I0718 16:29:23.099982   24101 rbd_util.go:229] rbd: map mon 10.44.XXX.YYY:6789
Jul 18 16:29:23 node-10-44-XXX-YYY origin-node: I0718 16:29:23.145500   24101 rbd_util.go:240] rbd: map error exit status 30 rbd: add failed: (30) Read-only file system
Jul 18 16:29:23 node-10-44-XXX-YYY origin-node: E0718 16:29:23.145555   24101 rbd.go:208] rbd: failed to setup
Jul 18 16:29:23 node-10-44-XXX-YYY origin-node: E0718 16:29:23.145629   24101 kubelet.go:1799] Unable to mount volumes for pod "hawkular-cassandra-1-x0shs_openshift-infra(fe2fd358-4ce2-11e6-9a04-001dd8f01d96)": rbd: map failed exit status 30 rbd: add failed: (30) Read-only file system
##### Expected Result

RBD map succeeds

**Comment by jameswilkins** [comment]
N.B - Ceph secret key is intentionally blank above

**Comment by copejon** [comment]
@jameswilkins I was able to work around this by editing /etc/systemd/system/atomic-openshift-node.service.  In the `ExecStart=/usr/bin/docker run ....` line there is a long list of -v options.   You'll need to change the `-v /sys:/sys:ro` to `-v /sys:/sys:rw`. 

Reload the daemon and restart the aos node service:

`systemctl daemon-reload`
`systemctl restart atomic-openshift-node`

This seems to be a bug.  I've submitted a BZ on this at https://bugzilla.redhat.com/show_bug.cgi?id=1367937
